### PR TITLE
Finishing translation of bnc poly

### PR DESCRIPTION
**Pull request by mariari**

### CODE REVIEW

File: src/geb/trans.lisp [hunk 1]
```diff
@@ -50,8 +50,20 @@
                                        (poly:+ (poly:* cz xin)
                                                (poly:- yzin cy)
                                                (poly:* cx cy)))))))
-    (pair          (error "not implemented"))
-    (case          (let ((obj-to-nat ))))
+    (pair          (let* ((z  (codom (mcdr obj)))
+                          (cz (obj-to-nat z)))
+                     (poly:* cz (poly:+ (to-poly (mcar obj))
+                                        (to-poly (mcdr obj))))))
```
Comment (mariari) [inline src/geb/trans.lisp:56]: This should match

```idris
substMorphToBNC (SMPair {x} {y} {z} f g) with (substObjToNat y, substObjToNat z)
  substMorphToBNC (SMPair {x} {y} {z} f g) | (cy, cz) =
    #| cz #* substMorphToBNC f #+ substMorphToBNC g
```

File: src/geb/trans.lisp [hunk 2]
```diff
@@ -50,8 +50,20 @@
...
+    (case          (let* ((f      (mcar obj))
+                          (x      (dom f))
+                          (cx     (obj-to-nat x))
+                          (poly-g (to-poly (mcadr obj))))
+                     (if (zerop cx)
+                         poly-g
+                         (poly:if-lt poly:ident cx
+                                     (to-poly f)
+                                     (poly:compose poly-g
+                                                   (poly:- poly:ident cx))))))
```
Comment (mariari) [inline src/geb/trans.lisp:66]: This should match

```idris
substMorphToBNC (SMCase {x} {y} {z} f g) with (substObjToNat x)
  substMorphToBNC (SMCase {x} {y} {z} f g) | cx =
    if cx == 0 then
      substMorphToBNC g
    else
      IfLT PI (#| cx)
        (substMorphToBNC f)
        (substMorphToBNC g #. (PI #- #| cx))
```

File: src/lambda/lambda-conversion.lisp [hunk 3]
```diff
@@ -11,47 +11,50 @@
...
+       (comp (<-left lty rty) (compile-checked-term context (prod lty rty) value)))
+      ((snd lty rty value)
+       (assert (eql (class-of rty) (class-of type)) nil "Types should match on fst: ~A ~A"
+               term type)
+       (comp (<-right lty rty) (compile-checked-term context (prod lty rty) value)))
+      ((lamb vty tty term)
+
+       (curry (commutes-left
+               ;; is this correct?
+               (compile-checked-term (cons vty context) tty term))))
```
Comment (mariari) [inline src/lambda/lambda-conversion.lisp:49]: this should match

```idris
compileCheckedTerm
  {ctx} {ty=(vty !-> tty)} (Checked_STLC_Lambda {vty} {tty} t) =
    soCurry $ soProdCommutesLeft $ compileCheckedTerm t
```

@rokopt can you confrim?

**Review by agureev — APPROVED** [review]
Change up `obj` to `mor` , otherwise all function translations are fine

File: src/geb/trans.lisp (hunk 1 above)
Comment (mariari) [reply src/geb/trans.lisp:56]: > So in CL presentation we do not have the constant constructor for BNCPolyM?

Correct, the sum type just has `fixnum` as a variant, and thus it all types out. The power of anonymous sum types a gogo!

**Pull request merged**